terraform/aks-cluster/main.tf

```diff
@@ -36,16 +37,19 @@ locals {
 
 resource "azurerm_kubernetes_cluster" "adl_aks" {
 
-  name                = "aks-${var.basename}"
-  location            = var.location
-  resource_group_name = var.rg_name
-  tags                = var.tags
-  dns_prefix          = var.dns_prefix
+  name                    = "aks-${var.basename}"
+  location                = var.location
+  resource_group_name     = var.rg_name
+  tags                    = var.tags
+  dns_prefix              = var.dns_prefix
+  private_cluster_enabled = var.is_sec_module && var.private_cluster_enabled ? true : false
```

Comment:
**[tfsec]** <[AVD-AZU-0041](https://aquasecurity.github.io/tfsec/v1.28.1/checks/azure/container/limit-authorized-ips/)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Cluster does not limit API access to specific IP addresses.